### PR TITLE
towersofhanoi

### PR DESCRIPTION
**Pull request by bir87**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

**Comment by bir87** [comment]
implementing returns

File: 03week/towersOfHanoi.js [hunk 1]
```diff
@@ -13,36 +13,67 @@ let stacks = {
...
 
 }
 
 function checkForWin() {
   // Your code here
+  return (stacks.b.length == 4 || stacks.c.length == 4);
+}
+
+function checkForValidInput(startStack, endStack){
+  return (typeof startStack === "string"|| typeof endStack === "string");
```
Comment (reneemeyer) [inline 03week/towersOfHanoi.js:47]: unnecessary parentheses

File: 03week/towersOfHanoi.js [hunk 2]
```diff
@@ -13,36 +13,67 @@ let stacks = {
...
+
+}
 
+function checkForOneCharInput(startStack, endStack){
+  const tester = /^[a-z]+$/;
+  if (!tester.test(startStack) || !tester.test(endStack)){
```
Comment (reneemeyer) [inline 03week/towersOfHanoi.js:53]: just return the evaluation

File: 03week/towersOfHanoi.js [hunk 3]
```diff
@@ -13,36 +13,67 @@ let stacks = {
...
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
+  
+  if (checkForValidInput(startStack,endStack)){
+    //checkForOneCharInput(startStack, endStack);
+    isLegal(startStack,endStack);
+    //console.log("afterislegal",startStack,endStack);
+    movePiece(startStack,endStack);
```
Comment (reneemeyer) [inline 03week/towersOfHanoi.js:64]: you should only move the piece if the move is legal

File: 03week/towersOfHanoi.js [hunk 4]
```diff
@@ -13,36 +13,67 @@ let stacks = {
...
+    //console.log("aftermovepiece");
+    checkForWin(startStack);
+    return checkForWin(startStack);
```
Comment (reneemeyer) [inline 03week/towersOfHanoi.js:67]: why are you running this twice?

**Review by reneemeyer — COMMENTED** [review]
see comments below

**Pull request closed without merging**